### DISCUSSION TITLE
调用智谱AI出错问题，聊天助手类型中工作流编排可用，但使用工作流类型就报错。

### DISCUSSION THREAD

**Discussion by ixujue** [langgenius/dify#3579] [Help] [answered]
如题，在聊天助手类型中的工作流编排情况下，智谱AI模型正常，使用工作流类型就会报错。一共4个可选模型，在工作流类型中只有1个模型能正常工作(chatglm_turbo)，两个报错(glm-3-turbo, glm-4)，还有1个是处于自娱自乐状态(glm-4v)，总是产生幻觉结果。 有两个提示出错：Node LLM run failed: [zhipuai] Error: Error code: 400, with error text {"error":{"code":"1214","message":"messages 参数非法。请检查文档。"}} 出这个错误时，可用的那个没问题，选另外两个就会报错。
![不可用模型的报错结果](https://github.com/langgenius/dify/assets/48663394/5ed688cc-e60f-4d58-9d22-659f470d65f8)
![不可用模型的幻觉结果](https://github.com/langgenius/dify/assets/48663394/34502c8e-7474-443c-b176-e2ac1d42c3b2)
![可用模型](https://github.com/langgenius/dify/assets/48663394/4f8c01a1-12ff-4ba1-ae60-a77f67d05734)
![可用模型的正常结果01](https://github.com/langgenius/dify/assets/48663394/66ffe46d-eac8-4202-a859-87bceec2a8b0)
![可用模型的正常结果02](https://github.com/langgenius/dify/assets/48663394/01dae1af-4a7e-4b01-8b75-2dec93d53bcd)

**Comment by crazywoola** [comment]
看看 Guideline 需要用英语提问题，截图可以是中文

**Comment by ferdaiyuan** [comment]
我遇到同样问题，解决了吗？

**Comment by ouyanmei** [comment]
我是把user的要求放到了system那里

**Comment by ouyanmei** [comment]
好像又是因为忘了加user这个message

**Comment by gnaij1106** [comment]
一样的报错，请问最后怎么解决的

**Comment by ouyanmei** [comment, marked as answer]
在message添加user，就是你的role不能只有system




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "langgenius/dify"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年7月26日(星期五) 中午11:38
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [langgenius/dify] 调用智谱AI出错问题，聊天助手类型中工作流编排可用，但使用工作流类型就报错。 (Discussion #3579)





 
一样的报错，请问最后怎么解决的

**Reply by TonyD0g** [reply]
感谢，这个是正解。

**Reply by Simon-wu** [reply]
我也是这么解决的，用了两个礼拜又抽风不行了，加了user
